addons/common/functions/fnc_rotateVectorAroundVector.sqf

```diff
@@ -0,0 +1,38 @@
+/*
+ * Author: LorenLuke
+ * Rotates the first vector around the second, clockwise by theta
+ *
+ * Arguments:
+ * 0: Vector <ARRAY>
+ * 1: Rotation Axis <ARRAY>
+ * 2: Angle <NUMBER>
+ *
+ * Return Value:
+ * Transformed Vector <ARRAY>
+ *
+ * Example:
+ * [eyeDirection player, weaponDirection (currentWeapon player), 45] call ace_common_fnc_rotateVectorAroundVector;
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vector1", "_vector2", "_theta"];
+_vector1 params ["_vx", "_vy", "_vz"];
+
+private _normalVector = vectorNormalized _vector2;
+_normalVector params ["_ux", "_uy", "_uz"];
+
+private _rotationMatrix = [
+    [cos(_theta) + ((_ux^2) * (1 - cos(_theta))), (_ux * _uy * (1-cos(_theta))) - (_uz * sin(_theta)), (_ux * _uz * (1 - cos(_theta))) + (_uy * sin (_theta))],
```

Comment:
You are doing `cos(_theta)` and `sin(_theta)` a lot in here. Saving it into a variable and reusing it would be alot more efficient.